main/manager.c

```diff
@@ -6873,6 +6901,175 @@ static int action_coreshowchannels(struct mansession *s, const struct message *m
 	return 0;
 }
 
+/*! \brief Helper function to add a channel name to the vector */
+static int coreshowchannelmap_add_to_map(struct ao2_container *c, const char *s)
+{
+	char *str;
+
+	str = ast_strdup(s);
+	if (!str) {
+		ast_log(LOG_ERROR, "Unable to append channel to channel map\n");
+		return 1;
+	}
+
+	/* If this is a duplicate, it will be ignored */
+	ast_str_container_add(c, str);
+
+	return 0;
+}
+
+/*! \brief Recursive function to get all channels in a bridge. Follow local channels as well */
+static int coreshowchannelmap_add_connected_channels(struct ao2_container *channel_map,
+	struct ast_channel_snapshot *channel_snapshot, struct ast_bridge_snapshot *bridge_snapshot)
+{
+	int res = 0;
+	struct ao2_iterator iter;
+	char *current_channel_uid;
+
+	iter = ao2_iterator_init(bridge_snapshot->channels, 0);
+	while ((current_channel_uid = ao2_iterator_next(&iter))) {
+		struct ast_channel_snapshot *current_channel_snapshot;
+		int add_channel_res;
+
+		/* Don't add the original channel to the list - it's either already in there,
+		 * or it's the channel we want the map for */
+		if (!strcmp(current_channel_uid, channel_snapshot->base->uniqueid)) {
+			continue;
+		}
+
+		current_channel_snapshot = ast_channel_snapshot_get_latest(current_channel_uid);
+		if (!current_channel_snapshot) {
+			ast_log(LOG_ERROR, "Unable to get snapshot of channel\n");
+			res = 1;
+			break;
+		}
+
+		add_channel_res = coreshowchannelmap_add_to_map(channel_map, current_channel_snapshot->base->name);
+		if (add_channel_res) {
+			res = 1;
+			ao2_ref(current_channel_snapshot, -1);
+			break;
+		}
+
+		/* If this is a local channel that we haven't seen yet, let's go ahead and find out what else is connected to it */
+		if (!add_channel_res && ast_begins_with(current_channel_snapshot->base->name, "Local")) {
```

Comment:
add_channel_res can never be non-zero here

Reply:
If add_channel_res is non-zero then at line 6948 you break out of the loop so line 6955 wouldn't execute. It's only if add_channel_res is zero that it could ever occur.